### PR TITLE
Closes #6344: Use activityName instead of packageName as primary key.

### LINKED ISSUE #6344: RecentApp should use activity name as a primary key
Using `packageName` as a primary key is not viable, as multiple options in share options could have the same package name. For example, "Save to drive" and "Copy to clipboard" both have `com.google.android.apps.docs`.
This leads to cases where options could be added to recent apps list in Fenix if another option with same package name has been used.

### PR DESCRIPTION
**Pull request by mcarare**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [x] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

https://github.com/mozilla-mobile/fenix/pull/9317

### CODE REVIEW

**Comment by mcarare** [comment]
> As we are changing the structure of the database, [we need to increase the database version number](https://github.com/mozilla-mobile/android-components/blob/260d9b43ebe00da2b0bf929fbacc2fbc84f67e83/components/feature/share/src/main/java/mozilla/components/feature/share/db/RecentAppsDatabase.kt#L15) and add a migration for deleting the `packageName` column.

I'not sure if we can migrate de current entries, because we might not know the activityName that corresponds to the existing packageNames.

**Comment by Amejia481** [comment]
> > As we are changing the structure of the database, [we need to increase the database version number](https://github.com/mozilla-mobile/android-components/blob/260d9b43ebe00da2b0bf929fbacc2fbc84f67e83/components/feature/share/src/main/java/mozilla/components/feature/share/db/RecentAppsDatabase.kt#L15) and add a migration for deleting the `packageName` column.
> 
> I'not sure if we can migrate de current entries, because we might not know the activityName that corresponds to the existing packageNames.

As we are creating a new field on the table(`activityName`) the old one (`packageName`) will be remained there, what I'm suggesting is in the migration code is to drop the `packageName` field as it contains information that we are not going to use anymore.

File: .buildconfig.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-componentsVersion: 37.0.0
+componentsVersion: 37.0.20200322190046
```
Comment (Amejia481) [inline .buildconfig.yml]: not sure if we want to include this change

Comment (mcarare) [reply .buildconfig.yml]: I did not :)

File: components/feature/share/src/main/java/mozilla/components/feature/share/db/RecentAppsDatabase.kt [hunk 2]
```diff
@@ -31,9 +35,28 @@ internal abstract class RecentAppsDatabase : RoomDatabase() {
...
             }
         }
     }
 }
+
+internal object Migrations {
+    val migration_1_2 = object : Migration(1, 2) {
+        override fun migrate(database: SupportSQLiteDatabase) {
+
+            database.execSQL("DROP TABLE RECENT_APPS_TABLE")
```
Comment (Amejia481) [inline components/feature/share/src/main/java/mozilla/components/feature/share/db/RecentAppsDatabase.kt]: 👍

**Comment by Amejia481** [comment]
> > > As we are changing the structure of the database, [we need to increase the database version number](https://github.com/mozilla-mobile/android-components/blob/260d9b43ebe00da2b0bf929fbacc2fbc84f67e83/components/feature/share/src/main/java/mozilla/components/feature/share/db/RecentAppsDatabase.kt#L15) and add a migration for deleting the `packageName` column.
> > 
> > 
> > I'not sure if we can migrate de current entries, because we might not know the activityName that corresponds to the existing packageNames.
> 
> As we are creating a new field on the table(`activityName`) the old one (`packageName`) will be remained there, what I'm suggesting is in the migration code is to drop the `packageName` field as it contains information that we are not going to use anymore.

Go it, the whole entries becomes invalid, I thought we have info that we could reuse 👍

**Pull request merged**